src/entities/asset/AssetIcon.vue

```diff
@@ -0,0 +1,69 @@
+<template>
+  <div class="asset">
+    <div class="asset__icon">
+      <XOR v-if="nameLower=='xor'" />
+      <VAL v-else-if="nameLower=='val'" />
+      <ETH v-else-if="nameLower=='eth'" />
+      <IrohaIcon v-else-if="nameLower=='iroha'" />
+      <Soshiba v-else-if="nameLower=='soshiba'" />
+      <Pswap v-else-if="nameLower=='pswap'" />
+      <NaIcon v-else />
+    </div>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { toRefs } from 'vue';
+
+import XOR from '~icons/assets/XOR.svg';
+import VAL from '~icons/assets/VAL.svg';
+import ETH from '~icons/assets/ETH.svg';
+import IrohaIcon from '~icons/assets/iroha_icon.svg';
+import Soshiba from '~icons/assets/soshiba.svg';
+import Pswap from '~icons/assets/pswap.svg';
+import NaIcon from '~icons/assets/na.svg';
```

Comment:
It might make sense to import these icons lazily, as dynamic imports. Also you can leverage [globs](https://vitejs.dev/guide/features.html#glob-import) to reduce boilerplate.